### PR TITLE
cheroot: bump to 8.5.2 and drop -27

### PR DESCRIPTION
**Pull request by alarcher**

### CODE REVIEW

File: components/python/cheroot/Makefile [hunk 1]
```diff
@@ -12,38 +12,34 @@
...
-COMPONENT_LICENSE_FILE=	LICENSE.md
-
-PYTHON_VERSIONS=	2.7 3.5
-
-COMPONENT_POST_INSTALL_ACTION = \
-        (cd $(PROTO_DIR)/usr/bin ; $(MV) -f cheroot cheroot-$(PYTHON_VERSION))
+COMPONENT_NAME=         cheroot
+COMPONENT_VERSION=      8.5.2
+COMPONENT_SUMMARY=      'Cheroot - high-performance, pure-Python HTTP server'
+COMPONENT_PROJECT_URL=  https://github.com/cherrypy/cheroot
```
Comment (AndWac) [inline components/python/cheroot/Makefile:23]: We should also remove the single quotes here.

File: components/python/cheroot/history [hunk 2]
```diff
@@ -1 +1,2 @@
+library/python/cheroot-27@6.5.2-2020.0.1.5
 library/python/cheroot-34@6.5.2-2020.0.1.2
```
Comment (AndWac) [inline components/python/cheroot/history:2]: Bot lines missing ",5.11", like
library/python/cheroot-27@6.5.2,5.11-2020.0.1.5

**Comment by AndWac** [comment]
Hm, local update doesn't work here:
pkg update: The requested change to the system attempts to install multiple actions
for link 'usr/bin/cheroot' with conflicting attributes:

    1 package delivers 'link mediator=python mediator-priority=vendor mediator-version=2.7 path=usr/bin/cheroot target=cheroot-2.7':
        pkg://openindiana.org/library/python/cheroot-27@6.5.2,5.11-2020.0.1.4:20201122T170118Z
    1 package delivers 'link mediator=python3 mediator-priority=vendor mediator-version=3.9 path=usr/bin/cheroot target=cheroot-3.9':
        pkg://openindiana.org/library/python/cheroot-39@8.5.2,5.11-2020.0.1.0:20210806T212853Z
    1 package delivers 'link mediator=python3 mediator-version=3.5 path=usr/bin/cheroot target=cheroot-3.5':
        pkg://openindiana.org/library/python/cheroot-35@8.5.2,5.11-2020.0.1.0:20210806T212850Z
    1 package delivers 'link mediator=python3 mediator-version=3.7 path=usr/bin/cheroot target=cheroot-3.7':
        pkg://openindiana.org/library/python/cheroot-37@8.5.2,5.11-2020.0.1.0:20210806T212852Z


These packages cannot be installed together. Any non-conflicting subset
of the above packages can be installed.

File: components/python/cheroot/history (hunk 2 above)
Comment (alarcher) [reply components/python/cheroot/history:2]: Again we should check but I was pretty sure we modified pkg to support short fmris...

**Comment by alarcher** [comment]
> Hm, local update doesn't work here:
> pkg update: The requested change to the system attempts to install multiple actions
> for link 'usr/bin/cheroot' with conflicting attributes:
> 
> ```
> 1 package delivers 'link mediator=python mediator-priority=vendor mediator-version=2.7 path=usr/bin/cheroot target=cheroot-2.7':
>     pkg://openindiana.org/library/python/cheroot-27@6.5.2,5.11-2020.0.1.4:20201122T170118Z
> 1 package delivers 'link mediator=python3 mediator-priority=vendor mediator-version=3.9 path=usr/bin/cheroot target=cheroot-3.9':
>     pkg://openindiana.org/library/python/cheroot-39@8.5.2,5.11-2020.0.1.0:20210806T212853Z
> 1 package delivers 'link mediator=python3 mediator-version=3.5 path=usr/bin/cheroot target=cheroot-3.5':
>     pkg://openindiana.org/library/python/cheroot-35@8.5.2,5.11-2020.0.1.0:20210806T212850Z
> 1 package delivers 'link mediator=python3 mediator-version=3.7 path=usr/bin/cheroot target=cheroot-3.7':
>     pkg://openindiana.org/library/python/cheroot-37@8.5.2,5.11-2020.0.1.0:20210806T212852Z
> ```
> 
> These packages cannot be installed together. Any non-conflicting subset
> of the above packages can be installed.

Right, need an optional dependency to trigger -27 deinstallation.

File: components/python/cheroot/Makefile (hunk 1 above)
Comment (alarcher) [reply components/python/cheroot/Makefile:23]: Fixed.

**Comment by alarcher** [comment]
> > Hm, local update doesn't work here:
> > pkg update: The requested change to the system attempts to install multiple actions
> > for link 'usr/bin/cheroot' with conflicting attributes:
> > ```
> > 1 package delivers 'link mediator=python mediator-priority=vendor mediator-version=2.7 path=usr/bin/cheroot target=cheroot-2.7':
> >     pkg://openindiana.org/library/python/cheroot-27@6.5.2,5.11-2020.0.1.4:20201122T170118Z
> > 1 package delivers 'link mediator=python3 mediator-priority=vendor mediator-version=3.9 path=usr/bin/cheroot target=cheroot-3.9':
> >     pkg://openindiana.org/library/python/cheroot-39@8.5.2,5.11-2020.0.1.0:20210806T212853Z
> > 1 package delivers 'link mediator=python3 mediator-version=3.5 path=usr/bin/cheroot target=cheroot-3.5':
> >     pkg://openindiana.org/library/python/cheroot-35@8.5.2,5.11-2020.0.1.0:20210806T212850Z
> > 1 package delivers 'link mediator=python3 mediator-version=3.7 path=usr/bin/cheroot target=cheroot-3.7':
> >     pkg://openindiana.org/library/python/cheroot-37@8.5.2,5.11-2020.0.1.0:20210806T212852Z
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > These packages cannot be installed together. Any non-conflicting subset
> > of the above packages can be installed.
> 
> Right, need an optional dependency to trigger -27 deinstallation.

Let me known if the optional dependency helps. I could install on my system but maybe the order of publishers has an influence.

**Comment by AndWac** [comment]
This time only cheroot-35 got updated, but cheroot and cheroot-27 stayed at its old version:
╰─➤  pkg list -a|grep cheroot
library/python/cheroot                            6.5.2-2020.0.1.4           i--
library/python/cheroot-27                         6.5.2-2020.0.1.4           i--
library/python/cheroot-34                         6.5.2-2020.0.1.2           --o
library/python/cheroot-35                         8.5.2-2020.0.1.0           i--
library/python/cheroot-37                         8.5.2-2020.0.1.0           ---
library/python/cheroot-39                         8.5.2-2020.0.1.0           ---

**Comment by alarcher** [comment]
> This time only cheroot-35 got updated, but cheroot and cheroot-27 stayed at its old version:
> ╰─➤ pkg list -a|grep cheroot
> library/python/cheroot 6.5.2-2020.0.1.4 i--
> library/python/cheroot-27 6.5.2-2020.0.1.4 i--
> library/python/cheroot-34 6.5.2-2020.0.1.2 --o
> library/python/cheroot-35 8.5.2-2020.0.1.0 i--
> library/python/cheroot-37 8.5.2-2020.0.1.0 ---
> library/python/cheroot-39 8.5.2-2020.0.1.0 ---

Do you think I should rather have an optional dependency to cheroot@the_version_that_deprecates_27 in cheroot-PYVER.p5m? I never remember how this works...

**Comment by AndWac** [comment]
I am not sure, either. Maybe toasterson has an idea.

**Comment by balrog-nona** [comment]
When I build it locally and try to install it, I get this error message
```
pkg install: No matching version of library/python/cheroot can be installed:
  Reject:  pkg://userland/library/python/cheroot@8.5.2-2020.0.1.0
  Reason:  No version matching 'optional' dependency library/python/cheroot-27@6.5.2-2020.0.1.5 can be installed
    ----------------------------------------
    Reject:  pkg://userland/library/python/cheroot-27@6.5.2-2020.0.1.5
    Reason:  Higher ranked publisher openindiana.org was selected
```

I don't know how to fix it in one step but I wonder if updating, adding Python3.7 and 3.9 and removing Python2.7 isn't too much for one step. I would split it into two: first PR with just obsoleting Python2.7 (increasing COMPONENT_REVISION and adding line into history file) and when it's merged, second PR with updating version plus adding new Pythons.

And isn't it problematic that cherrypy-27 according to .p5m file https://github.com/OpenIndiana/oi-userland/blob/25680651fe9872f5efc0f26ea536c45ba6e4e42c/components/python/cherrypy/cherrypy-PYVER.p5m#L26 still depends on cheroot-27? Shouldn't be cherrypy touched first?

**Comment by Toasterson** [comment]
Historically we where only able to drop old py versions when we dropped python together with everything. I think our dependency system is making too many assumptions

**Comment by AndWac** [comment]
Replaced by #7036

**Pull request closed without merging**